### ISSUE TITLE
[Copy PlainText] option to remove line breaks

### ISSUE THREAD

**Issue by wokier** [erosman/support#198]
The same way you allow to remove Line indent by key combination or permanent option, that would help me to be able to remove line breaks.
thanks

**Comment by erosman** [comment]
Currently, **Copy PlainText** removes multiple  line breaks and only keeps one.
e.g.:
```
This is line one.


This is line two.
```

Above is copied as:
```
This is line one.
This is line two.
```

However if lines have other characters e,g `line-break + space`, that won't happen.

Is that what you are referring to? or do you mean removing line breaks altogether so it becomes a single line like:

```
This is line one.This is line two.
```

**Comment by wokier** [comment]
good to know about the multi line break, but what i am looking for is the later you described (ie single line break).

**Comment by erosman** [comment]
> what i am looking for is the later you described (ie single line break).

Do you mean single line with no line breaks (_since single line break is already part of what **Copy PlainText** does_)?

If you mean no-line-break (one line text) since it is a very specific request, lets see if there are others who would want to have such feature.

**Comment by sbstn87** [comment]
Yeah, it would be great to have that functionality. It would be great if it could work like to the tool here:
https://www.textfixer.com/tools/remove-line-breaks.php

I use it all the time when copying text from PDFs. Having it available in your extension would make it much easier.

**Comment by erosman** [comment]
@sbstn87 Latest version (1.13) has custom Find & Replace that you can set to remove lines, replace and more.

**Comment by sbstn87** [comment]
Yeah, I have been trying many different ways to remove line breaks using RegEx, but is does not work for me. Perhaps someone here will let me know, what's wrong with the RegEx below:
Find: (\r\n)+|\r+|\n+|\t+/
Replace:  (that's a space)

**Comment by erosman** [comment]
**Find:** `/[\r\n]+/g`
**Replace:** (nothing)

`\t` is tab? Do you want to remove tab too?
Then ...
**Find:** `/[\r\n\t]+/g`

**Comment by sbstn87** [comment]
It works now perfectly. Thank you!

**Comment by sbstn87** [comment]
Update to the above - could you perhaps add this as a built-in option to the extension, just like removing line indents?

**Comment by erosman** [comment]
I made the Custom replace to avoid adding an option for each situation.

I was actually planning to remove "Remove Line Indent" option as well.

Let's see what works out best when I start working on **Copy PlainText** again.

**Comment by erosman** [comment]
Maybe, I create an interface to add the "most popular/used patterns" to the custom by clicking or selecting from a list. :thinking:

**Comment by sbstn87** [comment]
Yeah, this one sound like a great idea!

**Comment by erosman** [comment]
I am working on the next update of **Copy PlainText* and it will have ready-made find/replace patterns that users can add.

**Comment by sbstn87** [comment]
That's great to know. Thank you.

**Comment by erosman** [comment]
Let me know of any requests for ready-made patterns.

**Comment by sbstn87** [comment]
Some ideas:
- switch text to lower case
- switch text to UPPER CASE
- switch text to Title Case
- copy text without a provided word/words

**Comment by erosman** [comment]
> copy text without a provided word/words

That is easy and users can do so e.g.
- Pattern: `/(word1|word2|word3)/gi`
- Replacement:  _(nothing)_


> switch text to lower case .....

While changing case is easy in JavaScript, it is not possible with Regular Expression. 
_(I have another addon [Edit](https://addons.mozilla.org/firefox/addon/edit/) which does that)_
The custom function of the **Copy PlainText** only provides regular expression replace.

Furthermore, although I have included on/off feature for the patterns, they cant be selected at the time of copy. Therefore, any changes would occur all the time and making case changes may not be desired all the time.

![image](https://github.com/erosman/support/assets/6458876/c850fa50-f4e2-4132-849c-c148e1e8e36d)

**Comment by erosman** [comment]
I added a few more.
There is a special `(linkUrl)(linkText)`, which I needed myself. ;)

![image](https://github.com/erosman/support/assets/6458876/74d3d8f3-4b0c-405d-885b-fbca78b24bcf)

**Comment by sbstn87** [comment]
This looks great. Thank you for your work!
